### PR TITLE
release: closed beta features

### PR DESCRIPTION
**Pull request by kranthicodes**
## Summary
This PR contains features of closed-beta with new theme. 

- User wallet connection through arweave wallet kit
- User profiles
- Repository management
    - New Repo
    - Add files
    - Start issues
    - Start PRs
    - Create branches
    - Repo activity graph
- Search for repositories
- User activity heat map

## Tech stack (High level)
- React JS (TS) using Vite
- Tailwind css and headless ui for components and styling
- Zustand and immer for state management 
- Warp for contracts
- Isomorphic-git for git operations in browser

## Testing
Manually tested the above features

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,29 +5,83 @@
...
-    "arweave": "^1.13.7",
-    "bulma": "^0.9.4",
+    "@headlessui/react": "^1.7.16",
+    "@hookform/resolvers": "^3.2.0",
+    "@isomorphic-git/lightning-fs": "^4.6.0",
+    "@uiw/codemirror-extensions-langs": "^4.21.11",
+    "@uiw/codemirror-theme-github": "^4.21.11",
+    "@uiw/react-codemirror": "^4.21.11",
+    "@uiw/react-md-editor": "^3.23.5",
+    "arweave": "^1.14.4",
```
Comment (t8) [inline package.json:23]: Would ideally migrate to ArweaveKit in the future

File: index.html [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
   <head>
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <title>Protocol Land</title>
-    <script type="module">
-      import { Buffer } from "buffer";
-      window.Buffer = Buffer;
-    </script>
+    <title>Protocol Land - Decentralized git repositories storage</title>
```
Comment (t8) [inline index.html:7]:
```suggestion
    <title>Protocol Land - Beta</title>
```

File: src/assets/images/bg/banner-bg.svg
Comment (t8) [inline src/assets/images/bg/banner-bg.svg:1]: Is this still being used in the latest design?

File: src/assets/images/bg/left_bottom.png
Comment (t8) [inline src/assets/images/bg/left_bottom.png:1]: Is this still being used in the latest design?

File: src/assets/images/bg/right_bottom.png
Comment (t8) [inline src/assets/images/bg/right_bottom.png:1]: Is this still being used in the latest design?

File: src/assets/images/bg/right_top.png
Comment (t8) [inline src/assets/images/bg/right_top.png:1]: Is this still being used in the latest design?

File: src/assets/react.svg
Comment (t8) [inline src/assets/react.svg:1]: Is this still being used in the latest design?

File: index.html [hunk 3]
```diff
@@ -1,17 +1,13 @@
 <!DOCTYPE html>
-<html lang="en">
+<html data-color-mode="light" lang="en">
   <head>
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
```
Comment (t8) [inline index.html:5]: Can we update this to use a PL logo?

File: src/helpers/constants.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const CONTRACT_TX_ID = 'w5ZU15Y2cLzZlu3jewauIlnzbKw-OAxbN9G5TbuuiDQ'
+export const VITE_OTHENT_API_KEY = import.meta.env.VITE_OTHENT_API_KEY
```
Comment (t8) [inline src/helpers/constants.ts:2]: Will we need to remove ENV vars when the time comes to upload statically to Arweave?

File: src/lib/git/repo.ts [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+import { toArrayBuffer } from '@/helpers/toArrayBuffer'
+import { waitFor } from '@/helpers/waitFor'
+import { withAsync } from '@/helpers/withAsync'
+
+import { checkoutBranch, getCurrentBranch } from './branch'
+import { FSType } from './helpers/fsWithName'
+import { packGitRepo, unpackGitRepo } from './helpers/zipUtils'
+
+const arweave = new Arweave({
+  host: 'ar-io.net',
```
Comment (t8) [inline src/lib/git/repo.ts:18]: Can we verify with the ario team that this gateway is more stable than arweave.net?

File: src/pages/home/components/NewRepoModal.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+      const fs = fsWithName(title)
+      const createdRepo = await createNewRepo(title, fs, owner)
+
+      if (createdRepo && createdRepo.commit && createdRepo.repoBlob) {
+        const { repoBlob } = createdRepo
+
+        const result = await postNewRepo({ title, description, file: repoBlob, owner: authState.address })
+
+        if (result.id) {
+          trackGoogleAnalyticsEvent('Repository', 'Successfully create a repo', 'Create new repo', {
```
Comment (t8) [inline src/pages/home/components/NewRepoModal.tsx:64]:
```suggestion
          trackGoogleAnalyticsEvent('Repository', 'Successfully created a repo', 'Create new repo', {
```

File: package.json (hunk 1 above)
Comment (kranthicodes) [reply package.json:23]: Yes. I've created a task for it just now. https://communitylabs.atlassian.net/browse/PL-118?atlOrigin=eyJpIjoiMzNjNGMzNTY3ZTBiNGI4YjhjOTBkZTI3MGE0MzAyODgiLCJwIjoiaiJ9

File: src/lib/git/repo.ts (hunk 5 above)
Comment (kranthicodes) [reply src/lib/git/repo.ts:18]: Dropped a message asking their team about this.

**Review by t8 — APPROVED** [review]
LGTM!

**Pull request merged**